src/tools/perf/perftest_daemon.c

```diff
@@ -51,38 +51,25 @@ const char *ucp_perf_daemon_am_id_name(ucp_perf_daemon_am_id_t am_id)
     }
 }
 
-static void
-ucp_perf_daemon_ep_close(ucp_perf_daemon_context_t *ctx, ucp_ep_h ep)
+static void ucp_perf_daemon_ep_close(ucp_ep_h ep)
 {
     ucp_request_param_t param = {};
-    ucs_status_ptr_t close_req;
-    ucs_status_t status;
 
     if (NULL == ep) {
         return;
     }
 
+    ucs_debug("closing ep %p", ep);
     param.op_attr_mask = UCP_OP_ATTR_FIELD_FLAGS;
     param.flags        = UCP_EP_CLOSE_FLAG_FORCE;
-    close_req          = ucp_ep_close_nbx(ep, &param);
-
-    if (UCS_PTR_IS_PTR(close_req)) {
-        do {
-            ucp_worker_progress(ctx->worker);
-            status = ucp_request_check_status(close_req);
-        } while (status == UCS_INPROGRESS);
-        ucp_request_free(close_req);
-    } else {
-        status = UCS_PTR_STATUS(close_req);
-    }
-
-    if (status != UCS_OK) {
-        ucs_error("daemon failed to close ep %p", ep);
-    }
+    ucp_ep_close_nbx(ep, &param);
```

Comment:
why not waiting for the close request?

Reply:
I looked at the impl of `ucp_ep_close_nbx`. When we close with the `UCP_EP_CLOSE_FLAG_FORCE` flag, then returned request is just dummy:
```
    if (ucp_request_param_flags(param) & UCP_EP_CLOSE_FLAG_FORCE) {
        ucp_ep_discard_lanes(ep, UCS_ERR_CANCELED);
        ucp_ep_disconnected(ep, 1);
    } else {
        request = ucp_ep_flush_internal(ep, 0, param, NULL,
                                        ucp_ep_close_flushed_callback, "close");
```